pkg/minikube/service/service_test.go

```diff
@@ -493,4 +530,316 @@ func TestGetServiceURLsForService(t *testing.T) {
+		}
+	})
+	t.Run("ok no matches", func(t *testing.T) {
+		K8s = &MockClientGetter{
+			servicesMap:  serviceNamespaces,
+			endpointsMap: endpointNamespaces,
+		}
+		defer revertK8sClient(K8s)
+		ns, svc, label := "default", "mock-dashboard", "foo"
+		svcs, err := GetServiceListByLabel(ns, svc, label)
+		if err != nil {
+			t.Fatalf("GetServiceListByLabel not expected to fail but got err: %v. Namespace %v Service: %v, Label: %v", err, ns, svc, label)
+		}
+		if len(svcs.Items) == 1 {
+			t.Fatalf("GetServiceListByLabel for no matches should return empty list but got: %v", svcs.Items)
+		}
+	})
+
+	t.Run("ok", func(t *testing.T) {
+		K8s = &MockClientGetter{
+			servicesMap:  serviceNamespaces,
+			endpointsMap: endpointNamespaces,
+		}
+		defer revertK8sClient(K8s)
+		ns, svc, label := "default", "mock-dashboard", "mock"
+		svcs, err := GetServiceListByLabel(ns, svc, label)
+		if err != nil {
+			t.Fatalf("GetServiceListByLabel not expected to fail but got err: %v. Namespace %v Service: %v, Label: %v", err, ns, svc, label)
+		}
+		if len(svcs.Items) >= 1 {
+			t.Fatalf("GetServiceListByLabel for matching data should return at least 1 item but got: %v", svcs.Items)
+		}
+	})
+}
+
+func TestCheckService(t *testing.T) {
+	t.Run("svc-no-ports", func(t *testing.T) {
+		if err := CheckService("default", "mock-dashboard-no-ports"); err == nil {
+			t.Fatal("Error expected for service without ports")
+		}
+	})
+	t.Run("failed Get Client", func(t *testing.T) {
+		K8s = &MockClientGetter{
+			servicesMap:  serviceNamespaces,
+			endpointsMap: endpointNamespaces,
+		}
+		getCoreClientFail = true
+		defer revertK8sClient(K8s)
+		if err := CheckService("default", "mock-dashboard"); err == nil {
+			t.Fatal("Expected to fail for failed GetCoreClient")
+		}
+	})
+	t.Run("ok", func(t *testing.T) {
+		if err := CheckService("default", "mock-dashboard"); err != nil {
+			t.Fatalf("Unexpected error: %v while checking existing service", err)
+		}
+	})
+}
+
+func TestDeleteSecret(t *testing.T) {
+	t.Run("failed Get Client", func(t *testing.T) {
+		ns := "foo"
+		K8s = &MockClientGetter{
+			servicesMap:  serviceNamespaces,
+			endpointsMap: endpointNamespaces,
+			secretsMap:   secretsNamespaces,
+		}
+		getCoreClientFail = true
+		defer revertK8sClient(K8s)
+		if err := DeleteSecret(ns, "foo"); err == nil {
+			t.Fatal("CreateSecret expected to fail for failed GetCoreClient")
+		}
+	})
+	t.Run("ok", func(t *testing.T) {
+		ns := "foo"
+		secret := "mock-secret"
+		K8s = &MockClientGetter{
+			servicesMap:  serviceNamespaces,
+			endpointsMap: endpointNamespaces,
+			secretsMap:   secretsNamespaces,
+		}
+		defer revertK8sClient(K8s)
+		if err := DeleteSecret(ns, secret); err != nil {
+			t.Fatalf("DeleteSecret not expected to fail for namespace: %v and secret %v but got err: %v", ns, secret, err)
+		}
+	})
+}
+
+func TestCreateSecret(t *testing.T) {
+	t.Run("failed Get Client", func(t *testing.T) {
```

Comment:
Can this be refactored to use a table driven test?

https://github.com/golang/go/wiki/TableDrivenTests

Perhaps by adding a bool field for whether an error is expected?

Reply:
ok. Is table-driven approach preferred even if there are just 2 tests?